components/org.wso2.carbon.identity.application.authenticator.push.common/src/main/java/org/wso2/carbon/identity/application/authenticator/push/common/PushJWTValidator.java

```diff
@@ -168,4 +168,64 @@ private boolean isJWT(String tokenIdentifier) {
         }
     }
 
+    /**
+     * Get JWT claim from the claim set.
+     *
+     * @param claimsSet JWT claim set
+     * @param claim     Required claim
+     * @param deviceId  Device ID
+     * @return Claim string
+     * @throws PushAuthTokenValidationException if an error occurs while getting a claim
+     */
+    public String getClaimFromClaimSet(JWTClaimsSet claimsSet, String claim, String deviceId)
+            throws PushAuthTokenValidationException {
+
+        try {
+            return claimsSet.getStringClaim(claim);
+        } catch (ParseException e) {
+            String errorMessage = String.format("Failed to get %s from the auth response token received from device: "
+                    + "%s.", claim, deviceId);
+            throw new PushAuthTokenValidationException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Validate the correlation between the challenged saved in the server and received from the auth response.
+     *
+     * @param claimsSet JWT claim set for the validated auth response token
+     * @param challenge Challenge stored in cache to correlate with JWT
+     * @param deviceId  Unique ID for the device trying to authenticate the user
```

Comment:
Add @return to doc comment

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-push/pull/118/commits/061f61e9ccdcddd46a1a152bc5f126ed13cbd1aa